ddtrace/contrib/internal/google_generativeai/_utils.py

```diff
@@ -0,0 +1,219 @@
+            str(function_response_dict.get("name", "")),
+        )
+        span.set_tag_str(
+            "genai.request.contents.%d.parts.%d.function_response.response" % (content_idx, part_idx),
+            integration.trunc(str(function_response_dict.get("response", {}))),
+        )
+
+
+def _tag_request_content(span, integration, content, content_idx):
+    """Tag the generation span with request contents."""
+    if isinstance(content, str):
+        span.set_tag_str("genai.request.contents.%d.text" % content_idx, integration.trunc(content))
+        return
+    if isinstance(content, dict):
+        role = content.get("role", "")
+        if role:
+            span.set_tag_str("genai.request.contents.%d.role" % content_idx, str(content.get("role", "")))
+        parts = content.get("parts", [])
+        for part_idx, part in enumerate(parts):
+            _tag_request_content_part(span, integration, part, part_idx, content_idx)
+        return
+    role = getattr(content, "role", "")
+    if role:
+        span.set_tag_str("genai.request.contents.%d.role" % content_idx, str(role))
+    parts = getattr(content, "parts", [])
+    if not parts:
+        span.set_tag_str(
+            "genai.request.contents.%d.text" % content_idx,
+            integration.trunc("[Non-text content object: {}]".format(repr(content))),
+        )
+        return
+    for part_idx, part in enumerate(parts):
+        _tag_request_content_part(span, integration, part, part_idx, content_idx)
+
+
+def _tag_response_part(span, integration, part, part_idx, candidate_idx):
+    """Tag the generation span with response part text and function calls."""
+    text = part.get("text", "")
+    span.set_tag_str(
+        "genai.response.candidates.%d.content.parts.%d.text" % (candidate_idx, part_idx),
+        integration.trunc(str(text)),
+    )
+    function_call = part.get("function_call", None)
+    if not function_call:
+        return
+    span.set_tag_str(
+        "genai.response.candidates.%d.content.parts.%d.function_call.name" % (candidate_idx, part_idx),
+        integration.trunc(str(function_call.get("name", ""))),
+    )
+    span.set_tag_str(
+        "genai.response.candidates.%d.content.parts.%d.function_call.args" % (candidate_idx, part_idx),
+        integration.trunc(str(function_call.get("args", {}))),
+    )
+
+
+def tag_request(span, integration, instance, args, kwargs):
+    """Tag the generation span with request details.
+    Includes capturing generation configuration, system prompt, and messages.
+    """
+    contents = get_argument_value(args, kwargs, 0, "contents")
+    generation_config = kwargs.get("generation_config", {})
+    system_instruction = getattr(instance, "_system_instruction", None)
+    stream = kwargs.get("stream", None)
+
+    generation_config_dict = None
+    if isinstance(generation_config, dict):
+        generation_config_dict = generation_config
+    elif generation_config is not None:
```

Comment:
in the cases is `generation_config` not a dictionary? I see some examples have generation_config of type `genai.GenerationConfig`. Should we explicitly check for that instead?

This way if someone passes in an invalid type for `generation_config` we won't just be dumping the `.__dict__` attributes onto the span